src/main/kotlin/org/openrndr/panel/elements/PlanarPad.kt

```diff
@@ -0,0 +1,198 @@
+        val old = value
+
+        if (keyEvent.key == KEY_ARROW_RIGHT) {
+            value = Vector2(value.x + delta, value.y)
+        }
+
+        if (keyEvent.key == KEY_ARROW_LEFT) {
+            value = Vector2(value.x - delta, value.y)
+        }
+
+        if (keyEvent.key == KEY_ARROW_UP) {
+            value = Vector2(value.x, value.y - delta)
+        }
+
+        if (keyEvent.key == KEY_ARROW_DOWN) {
+            value = Vector2(value.x, value.y + delta)
+        }
+
+        draw.dirty = true
+        events.valueChanged.onNext(ValueChangedEvent(this, old, value))
+        keyEvent.cancelPropagation()
+    }
+
+    private fun pick(e: MouseEvent) {
+        val old = value
+
+        // Difference
+        val dx = e.position.x - layout.screenX
+        var dy = e.position.y - layout.screenY
+
+        // Normalize to -1 - 1
+        val nx = clamp(dx / layout.screenWidth * 2.0 - 1.0, -1.0, 1.0)
+        val ny = clamp(dy / layout.screenHeight * 2.0 - 1.0, -1.0, 1.0)
+
+        normalizedValue = Vector2(nx, ny)
+
+        events.valueChanged.onNext(ValueChangedEvent(this, old, value))
+        draw.dirty = true
+    }
+
+    override val widthHint: Double?
+        get() = 200.0
+
+
+    private val ballPosition: Vector2
+        get() = Vector2(
+                map(-1.0, 1.0, 0.0, layout.screenWidth, normalizedValue.x),
+                map(-1.0, 1.0, 0.0, layout.screenHeight, normalizedValue.y)
+        )
+
+    override fun draw(drawer: Drawer) {
+        computedStyle.let {
+            drawer.pushTransforms()
+            drawer.pushStyle()
+            drawer.fill = ColorRGBa.GRAY
+            drawer.stroke = null
+            drawer.strokeWeight = 0.0
+
+            drawer.rectangle(0.0, 0.0, layout.screenWidth, layout.screenHeight)
+
+
+            // lines grid
+            drawer.stroke = ColorRGBa.GRAY.shade(1.2)
+            drawer.strokeWeight = 1.0
+
+            for (y in 0 until 21) {
+                drawer.lineSegment(
+                        0.0,
+                        layout.screenHeight / 20 * y,
+                        layout.screenWidth - 1.0,
+                        layout.screenHeight / 20 * y
+                )
+            }
+
+            for (x in 0 until 21) {
+                drawer.lineSegment(
+                        layout.screenWidth / 20 * x,
+                        0.0,
+                        layout.screenWidth / 20 * x,
+                        layout.screenHeight - 1.0
+                )
+            }
+
+            // cross
+            drawer.stroke = ColorRGBa.GRAY.shade(1.6)
+//            drawer.lineSegment(0.0, layout.screenHeight / 2.0, layout.screenWidth, layout.screenHeight / 2.0)
+//            drawer.lineSegment(layout.screenWidth / 2.0, 0.0, layout.screenWidth / 2.0, layout.screenHeight)
+
+            // angle line from center
+            drawer.lineSegment(Vector2(layout.screenHeight / 2.0, layout.screenWidth / 2.0), ballPosition)
```

Comment:
What would be the purpose of this?

Reply:
Edwin suggested it, so people could use it for directions as well...

Reply:
Makes sense.. make it toggle-able?

Reply:
Alright! Should it be default on or off?

Reply:
I'd say off